rcos/Telescope
# Admin panel

Issue Number: close #110 

Problem Summary:
Create an admin panel for coordinators and faculty advisors to manage over-arching items in the database(i.e. semesters).

### What is changed and how it works?
**graphql/rcos/users/enrollments: "enrollments_lookup.graphql", "user_enrollment_lookup.graphql", src/api/rcos/users/enrollments:: "enrollments_lookup.rs", "user_enrollment_lookup.rs",
src/web/services/admin/semesters: "view_enrollments.rs"**

#### view_enrollments.rs:
- Implemented enrollment page for corresponding semester.
- Admin now can see who is enrolled in selected semester.
- Admin could search user in selected semester by first name, last name or rcs id.
- Implemented "export to csv" button, which can export enrollment data of selected semester to csv file.

### Tests

Manual test:
1. Open Hasura and insert 2 semesters(semester_id = 202201, 202101) record in semester table.
2. Create >20 number of users who enrolled in semester with semester_id = 202201.
3. Open ROCS website and login as coordinators/faculty advisors.
4. Click "Admin" button on the top.
5. Click "manage" and two semesters on the list.
6. Click on the semester with semester_id = 202201, and list of students are shown on the page.
7. Click the page number to see if pagination works.
8. Search a user by their first name/ last name/rcs id to see if search box works.
9. Click the "Export to CSV" button on the left, a new download page tab will be created and a csv file that includes enrollment data of selected semester will be downloaded in your computer, the name should be `<semester title>+_enrollments.csv`.


PS:
1. When user access to the download page of enrollments, a csv file will be created and written at /tmp/ with name `{semester_id}_enrollments.csv` in server. I don't know if there is any better way to handle it.
2. I am still trying to figure out how to auto close the download page once the download has finished. 
3. The csv file just simply includes all information in semester table in string format(except for related table), the null field is replace by empty string. If there is any additional requirement for exported csv file, please point out.
4. If more information is needed on admin panel page, please point out and I will make update.
